pwndbg/dbg/gdb.py

```diff
@@ -13,7 +15,41 @@
 from pwndbg.commands import load_commands
 from pwndbg.gdblib import gdb_version
 from pwndbg.gdblib import load_gdblib
-from pwndbg.gdblib import prompt
+
+
+class GDBSession(pwndbg.dbg_mod.Session):
+    @override
+    def history(self) -> List[str]:
+        lines = gdb.execute("show commands", from_tty=False, to_string=True)
+        return lines.splitlines()
+
+    @override
+    def lex_args(self, command_line: str) -> List[str]:
+        return gdb.string_to_argv(command_line)
+
+
+class GDBCommand(gdb.Command):
+    def __init__(
+        self,
+        debugger: GDB,
+        name: str,
+        handler: Callable[[pwndbg.dbg_mod.Debugger, str, bool], None],
+    ):
+        self.debugger = debugger
+        self.handler = handler
+        super().__init__(name, gdb.COMMAND_USER, gdb.COMPLETE_EXPRESSION)
+
+    def invoke(self, args: str, from_tty: bool) -> None:
+        self.handler(self.debugger, args, from_tty)
+
+
+class GDBCommandHandle(pwndbg.dbg_mod.CommandHandle):
+    def __init__(self, command: gdb.Command):
+        self.command = command
+
+    def remove(self) -> None:
+        # GDB doesn't support command removal?
```

Comment:
Pretty sure you're correct it doesn't, we can remove the question mark. BTW, we should file a bug for this if we haven't already (check with @disconnect3d).

Reply:
I believe we didn't.